src/main/java/ca/nines/ise/validator/SectionCoverageValidator.java

```diff
@@ -0,0 +1,176 @@
+      "validator.coverage.outside_body",
+      "validator.coverage.outside_page"
+  })
+  private void process_text(TextNode n){
+    //if text is empty, return
+    if (n.getText().trim().equals(""))
+      return;
+    //if not in matter
+    if (!nodeStack.in_tag("frontmatter") && !nodeStack.in_tag("backmatter")){
+      //if no sectioning exists
+      if (coverAll.isEmpty() && scenes.isEmpty() && acts.isEmpty() && pages.isEmpty()){
+        Message m = Message.builder("validator.coverage.required")
+          .fromNode(n)
+          .addNote("Text in the document must be within at least one type of sectioning element (ex. DIV, ACT, SCENE, etc.)")
+          .build();
+        Log.addMessage(m); 
+      }
+      //if not in a scene or act
+      if (!nodeStack.in_tag("act") && !nodeStack.in_tag("scene")){
+        //if acts or scenes exist
+        if (!scenes.isEmpty() || !acts.isEmpty()){
+          Message m = Message.builder("validator.coverage.outside_body")
+            .fromNode(n)
+            .addNote("All text outside FRONTMATTER and BACKMATTER must be within an ACT and/or SCENE if ACT and/or SCENE tags exist in the document")
+            .build();
+          Log.addMessage(m);
+        }
+      }
+    }else if (!nodeStack.in_tag("div")){
+      Message m = Message.builder("validator.coverage.outside_div")
+          .fromNode(n)
+          .addNote("Text in FRONTMATTER or BACKMATTER must be within a DIV")
+          .build();
+      Log.addMessage(m);
+    }
+    
+    Boolean outside = true;
+    if (coverAll.isEmpty())
+      outside = false;
+    else{
+      for (Node c : coverAll){
+        if (nodeStack.in_tag(c.getName()))
+           outside = false;
+      }
+    }
+    if (outside){
+      Message m = Message.builder("validator.coverage.outside_body")
+          .fromNode(n)
+          .addNote("All text in the document must be contained within DIV tags if a DIV exists outside FRONTMATTER and BACKMATTER")
+          .build();
+      Log.addMessage(m);
+    }
+    
+    //if not in a page but pages exist, error
+    if (!pages.isEmpty() && !nodeStack.in_tag("page")){
+      Message m = Message.builder("validator.coverage.outside_page")
+          .fromNode(n)
+          .addNote("if PAGE tags exists, all text must be within a PAGE")
+          .build();
+      Log.addMessage(m);
+    }
+    
+  }
+  
+  public void validate(DOM dom) {
+    nodeStack = new ValidatorStack<StartNode>();
+    coverAll = new ArrayList<StartNode>();
+    acts = new ArrayList<StartNode>();
+    scenes = new ArrayList<StartNode>();
+    pages = new ArrayList<StartNode>();
+    
+    //first pass
+    for (Node n : dom) {
+      switch (n.type()) {
+        case START:
+          process_start((StartNode) n);
+          process_init((StartNode) n);
+          break;
+        case END:
+          process_end((EndNode) n);
+          break;
+      }
+    }
+    
+    nodeStack = new ValidatorStack<StartNode>();
+    
+    //second pass
```

Comment:
Although, now that I look at my code I have no idea why I decided to put coverAll, acts, scenes and pages in lists. Those should just be booleans.

I suppose I could also just use dom.find_forward(). I guess that would probably be the easiest.